Unable to get `#[from]` + `backtrace` example to compile

<!--  ❤️  -->

I have a use case where this example from the README is exactly what I need:

```rust
#[derive(Error, Debug)]
pub enum MyError {
    Io {
        #[from]
        source: io::Error,
        backtrace: Backtrace,
    },
}
```

However, when I compile it ([Rust Playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3819922d82ecbafa0900ab66fc068a15)) I get the following error:

<details>
<summary>no `ThiserrorProvide` in `__private`</summary>
<pre>
   Compiling playground v0.0.1 (/playground)
error[E0432]: unresolved import `thiserror`
   --> src/lib.rs:5:10
    |
5   | #[derive(Error, Debug)]
    |          ^^^^^ no `ThiserrorProvide` in `__private`
    |
note: found an item that was configured out
   --> /playground/.cargo/registry/src/index.crates.io-6f17d22bba15001f/thiserror-1.0.48/src/lib.rs:256:29
    |
256 |     pub use crate::provide::ThiserrorProvide;
    |                             ^^^^^^^^^^^^^^^^
    = note: this error originates in the derive macro `Error` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0658]: use of unstable library feature 'error_generic_member_access'
 --> src/lib.rs:5:10
  |
5 | #[derive(Error, Debug)]
  |          ^^^^^
  |
  = note: see issue #99301 <https://github.com/rust-lang/rust/issues/99301> for more information
  = note: this error originates in the derive macro `Error` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: `MyError` doesn't implement `std::fmt::Display`
 --> src/lib.rs:6:10
  |
6 | pub enum MyError {
  |          ^^^^^^^ `MyError` cannot be formatted with the default formatter
  |
  = help: the trait `std::fmt::Display` is not implemented for `MyError`
  = note: in format strings you may be able to use `{:?}` (or {:#?} for pretty-print) instead
note: required by a bound in `std::error::Error`
 --> /rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/core/src/error.rs:32:1

</pre>
</details>

--- a/README.md
+++ b/README.md
@@ -88,20 +88,18 @@ pub enum DataStoreError {
   }
   ```
 
-- A `From` impl is generated for each variant containing a `#[from]` attribute.
+- A `From` impl is generated for each variant that contains a `#[from]`
+  attribute.
 
-  Note that the variant must not contain any other fields beyond the source
-  error and possibly a backtrace. A backtrace is captured from within the `From`
-  impl if there is a field for it.
+  The variant using `#[from]` must not contain any other fields beyond the
+  source error (and possibly a backtrace &mdash; see below). Usually `#[from]`
+  fields are unnamed, but `#[from]` is allowed on a named field too.
 
   ```rust
   #[derive(Error, Debug)]
   pub enum MyError {
-      Io {
-          #[from]
-          source: io::Error,
-          backtrace: Backtrace,
-      },
+      Io(#[from] io::Error),
+      Glob(#[from] globset::Error),
   }
   ```
 
@@ -125,7 +123,9 @@ pub enum DataStoreError {
   ```
 
 - The Error trait's `provide()` method is implemented to provide whichever field
-  has a type named `Backtrace`, if any, as a `std::backtrace::Backtrace`.
+  has a type named `Backtrace`, if any, as a `std::backtrace::Backtrace`. Using
+  `Backtrace` in errors requires a nightly compiler with Rust version 1.73 or
+  newer.
 
   ```rust
   use std::backtrace::Backtrace;
@@ -140,7 +140,9 @@ pub enum DataStoreError {
 - If a field is both a source (named `source`, or has `#[source]` or `#[from]`
   attribute) *and* is marked `#[backtrace]`, then the Error trait's `provide()`
   method is forwarded to the source's `provide` so that both layers of the error
-  share the same backtrace.
+  share the same backtrace. The `#[backtrace]` attribute requires a nightly
+  compiler with Rust version 1.73 or newer.
+
 
   ```rust
   #[derive(Error, Debug)]
@@ -148,6 +150,20 @@ pub enum DataStoreError {
       Io {
           #[backtrace]
           source: io::Error,
+      },
+  }
+  ```
+
+- For variants that use `#[from]` and also contain a `Backtrace` field, a
+  backtrace is captured from within the `From` impl.
+
+  ```rust
+  #[derive(Error, Debug)]
+  pub enum MyError {
+      Io {
+          #[from]
+          source: io::Error,
+          backtrace: Backtrace,
       },
   }
   ```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -98,24 +98,36 @@
 //!   }
 //!   ```
 //!
-//! - A `From` impl is generated for each variant containing a `#[from]`
+//! - A `From` impl is generated for each variant that contains a `#[from]`
 //!   attribute.
 //!
-//!   Note that the variant must not contain any other fields beyond the source
-//!   error and possibly a backtrace. A backtrace is captured from within the
-//!   `From` impl if there is a field for it.
+//!   The variant using `#[from]` must not contain any other fields beyond the
+//!   source error (and possibly a backtrace &mdash; see below). Usually
+//!   `#[from]` fields are unnamed, but `#[from]` is allowed on a named field
+//!   too.
 //!
 //!   ```rust
-//!   # const IGNORE: &str = stringify! {
+//!   # use core::fmt::{self, Display};
+//!   # use std::io;
+//!   # use thiserror::Error;
+//!   #
+//!   # mod globset {
+//!   #     #[derive(thiserror::Error, Debug)]
+//!   #     #[error("...")]
+//!   #     pub struct Error;
+//!   # }
+//!   #
 //!   #[derive(Error, Debug)]
 //!   pub enum MyError {
-//!       Io {
-//!           #[from]
-//!           source: io::Error,
-//!           backtrace: Backtrace,
-//!       },
+//!       Io(#[from] io::Error),
+//!       Glob(#[from] globset::Error),
 //!   }
-//!   # };
+//!   #
+//!   # impl Display for MyError {
+//!   #     fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+//!   #         unimplemented!()
+//!   #     }
+//!   # }
 //!   ```
 //!
 //! - The Error trait's `source()` method is implemented to return whichever
@@ -148,7 +160,8 @@
 //!
 //! - The Error trait's `provide()` method is implemented to provide whichever
 //!   field has a type named `Backtrace`, if any, as a
-//!   `std::backtrace::Backtrace`.
+//!   `std::backtrace::Backtrace`. Using `Backtrace` in errors requires a
+//!   nightly compiler with Rust version 1.73 or newer.
 //!
 //!   ```rust
 //!   # const IGNORE: &str = stringify! {
@@ -165,7 +178,8 @@
 //! - If a field is both a source (named `source`, or has `#[source]` or
 //!   `#[from]` attribute) *and* is marked `#[backtrace]`, then the Error
 //!   trait's `provide()` method is forwarded to the source's `provide` so that
-//!   both layers of the error share the same backtrace.
+//!   both layers of the error share the same backtrace. The `#[backtrace]`
+//!   attribute requires a nightly compiler with Rust version 1.73 or newer.
 //!
 //!   ```rust
 //!   # const IGNORE: &str = stringify! {
@@ -174,6 +188,22 @@
 //!       Io {
 //!           #[backtrace]
 //!           source: io::Error,
+//!       },
+//!   }
+//!   # };
+//!   ```
+//!
+//! - For variants that use `#[from]` and also contain a `Backtrace` field, a
+//!   backtrace is captured from within the `From` impl.
+//!
+//!   ```rust
+//!   # const IGNORE: &str = stringify! {
+//!   #[derive(Error, Debug)]
+//!   pub enum MyError {
+//!       Io {
+//!           #[from]
+//!           source: io::Error,
+//!           backtrace: Backtrace,
 //!       },
 //!   }
 //!   # };

